js/admission.js

```diff
@@ -0,0 +1,134 @@
+            {id:4, title: "User Experience Design", period: "Summer 2021", level: "Bachelor's", subjectArea: "Computer Science", paceOfStudy: "75%", teachingForm: "Course", applicationCode: "EUA-126"},
+            {id:5, title: "Evolutionary Biology", period: "Autumn 2021", level: "Master's", subjectArea: "Biology", paceOfStudy: "Full-time", teachingForm: "Course", applicationCode: "EUA-127"},
+            {id:6, title: "Cell and Developmental Biology", period: "Autumn 2021", level: "Bachelor's", subjectArea: "Biology", paceOfStudy: "Full-time", teachingForm: "Program", applicationCode: "EUA-128"},
+            {id:7, title: "Human Physiology", period: "Summer 2021", level: "Beginner", subjectArea: "Biology", paceOfStudy: "Full-time", teachingForm: "Course", applicationCode: "EUA-129"}
+        ];
+
+        return courses;
+    }
+
+    function populateUI(data){
+        $("#search-result").empty();
+        data.forEach(course => {
+            var html = `<div class="searchresult-card">
+            <div class="resultcard-header">
+            <h3 class="column-heading">${course.title}</h3>
+            </div>
+            <div class="resultcard-content font-light-color">
+            <div class="resultcard-content-col">
+            <p class="line-spacing"><span class="subtitle">Period: </span>${course.period}</p>
+            <p class="line-spacing"><span class="subtitle">Level: </span>${course.level}</p>
+            <p class="line-spacing"><span class="subtitle">Subject Area: </span>${course.subjectArea}</p>
+            </div>
+            <div class="resultcard-content-col">
+            <p class="line-spacing"><span class="subtitle">Pace of study: </span>${course.paceOfStudy}</p>
+            <p class="line-spacing"><span class="subtitle">Teaching form: </span>${course.teachingForm}</p>
+            <p class="line-spacing"><span class="subtitle">Application code: </span>${course.applicationCode}</p>
+            </div>
+            </div>
+            </div>`;
+            $("#search-result").append(html);
+        });
+    }
+
+    function displayerrorMessage(){
+        var html = `<div class="noMatchingResult searchresult-card">
+                            <div class="image-container"></div>
+                            <p class="subtitle text-center">Your search criteria did not match any courses or programs</p>
+                            </div>`;
+                            $("#search-result").append(html);
+    }
+
+// display result as per searched text and semester selection in search bar
+
+    var filteredList = [];
+    courseDataAfterSearch =  courseData;
+    $("#search").submit(function() {
+        var searchText = $("#freeText").val();
+        var semester = $("#period").val();
+        filteredList = courseData.filter(course => ((course.period.toUpperCase().indexOf(semester.toUpperCase()) != -1) && (course.subjectArea.toUpperCase().indexOf(searchText.toUpperCase()) != -1)));
```

Comment:
it's better for readability to define a few variables before using them in the complex condition
```suggestion
        const isMatchedBySemester = course.period.toUpperCase().indexOf(semester.toUpperCase()) != -1;
        const isMatchedBySubjectArea = course.subjectArea.toUpperCase().indexOf(searchText.toUpperCase()) != -1;
        
        filteredList = courseData.filter(course => isMatchedBySemester && isMatchedBySubjectArea);
```